src/odinapi/views/read_apriori.py

```diff
@@ -30,36 +34,38 @@ def get_data(key):
     }
 
 
+def get_interpolation_weights(
+    xs: np.array, x: float, doy_interpolation: bool = False
+) -> Tuple[int, int, float, float]:
+    if x <= xs[0] and not doy_interpolation:
+        return 0, xs.size - 1, 1., 0.
+    if x >= xs[-1] and not doy_interpolation:
+        return 0, xs.size - 1, 0., 1.
+    if (x >= xs[-1] or x <= xs[0]) and doy_interpolation:
+        dx = DAYS_PER_YEAR + xs[0] - xs[xs.size - 1]
+        xi = min(x, DAYS_PER_YEAR)  # neglect leap year
+        w1 = ((xi - xs[xs.size - 1]) % DAYS_PER_YEAR) / dx
+        return 0, xs.size - 1, w1, 1. - w1
+    ind2 = np.argmax(x <= xs)
+    ind1 = ind2 - 1
+    dx = xs[ind2] - xs[ind1]
+    w1 = (xs[ind2] - x) / dx
+    return ind1, ind2, w1, 1. - w1
```

Comment:
I don't quite understand this. Let me see if I can make it out:
1st case: beginning of year, `doy_interpolation` is `False`, return all weight on first value.
2nd case: end of year, `doy_interpolation` is `False`, return all weight on last value.
3rd case: either beginning or end, `doy_interpolation` is `True`, find interpolated value for this period (no need to use `xs.size - 1`, right? just `-1` should be enough?)
4th case: regardless of `doy_interpolation`, find interpolated value for rest of year.

Is this correct?

If so, why is there no corresponding return closest value for times that are not at the end/beginning of year?

And why do we need the `doy_intrepolatin` flag at all? Shouldn't we always interpolate?

Reply:
You understand it correct. We need the flag since we also use this for latitude interpolation, where we should not interpolate between "edges".
For the 4th case we are sure that we are not at the beginning or end the year, so I do not understand one of your question.

Reply:
The confusion was due to the dual use of this function. Since there was a case where you returned nearest, I thought there ought to be such a case also in mid year, that is to say I interpreted the flag being `False` to mean "give me nearest`. I'm not sure it's a good idea to have one function doing both these things, if I'm honest. As you can see, it confused me. :)

Reply:
Ok, then I will remove the logic and split it into three functions.

Reply:
If you agree. Three though? What's the third case?

Reply:
get_interpolation_weights_for_lat and get_interpolation_weights_for_doy and then get_interpolation_weights that the orher functions calls.